src/main/java/io/jenkins/plugins/signpath/SignPathStepBase.java

```diff
@@ -34,17 +39,43 @@ public abstract class SignPathStepBase extends Step {
     // also a timeout that is too high is not useful anymore
     private int waitForPowerShellTimeoutInSeconds = (int) TimeUnit.MINUTES.toSeconds(30);
 
-    private String trustedBuildSystemTokenCredentialId = "SignPath.TrustedBuildSystemToken";
+    private String trustedBuildSystemTokenCredentialId;
     private String apiTokenCredentialId = "SignPath.ApiToken";
 
     public String getApiUrl() {
         return apiUrl;
     }
 
+    // we use this method in the task to avoid overriding empty values at build level
+    // with the values from the global configuration
+    public String getApiUrlWithGlobalConfig() {
```

Comment:
All the **get<configValue>WithGlobalConfig** methods are very similar: they do a null or empty check and then retrieve it from global config. Can we extract this common logic into a parameterized method?

Reply:
extracted the common part to a separate method